packages/jest-core/src/cli/index.ts

```diff
@@ -145,6 +146,23 @@ const _run = async (
   // as soon as possible, so by the time we need the result it's already there.
   const changedFilesPromise = getChangedFilesPromise(globalConfig, configs);
 
+  // Filter may need to do an HTTP call or something similar to setup.
+  // We will not wait on an async response from this before using the filter.
+  let filter: Filter | undefined;
+  if (globalConfig.filter && !globalConfig.skipFilter) {
+    const rawFilter = require(globalConfig.filter);
+    let filterSetupPromise: Promise<void> | undefined;
+    if (rawFilter.setup) {
+      filterSetupPromise = rawFilter.setup();
```

Comment:
what happens if this promise rejects before `await filter` in `SearchSource`? Will that bubble up as unhandled?

Reply:
we could add a `.catch` here and store the rejection in a variable and throw it manually? not sure

Reply:
I just tested and the filter setup hook being thrown is caught in the same place that the filter itself is called and has all the same behavior as if the filter failed. Which, currently, means the filter is ignored.

Reply:
** I meant, not that the filter is ignored, but 0 results are returned.

If/when we add handling for the filter being thrown, the setup hook will be lumped in with it automatically.

Reply:
I'm adding tests for both filter and setup filter throwing so we at least know the behavior, although when it happens Jest won't run tests (and that's expected).

Reply:
Thanks for feedback; I did find a way to improve the behavior and I wrote tests for both filter and setup filter error code paths.

Reply:
Sounds great!